Bug 2021405: [e2e][automation] VM creation wizard Cloud Init editor

lkladnit opened the pull request:
Cypress tests for [CNV-14292: add TC/automation for fixed bugs](https://issues.redhat.com/browse/CNV-14292), [Polarion TC](https://polarion.engineering.redhat.com/polarion/#/project/CNV/workitem?id=CNV-7294)

VNC login tests are moved to tier 2: https://gitlab.cee.redhat.com/cnv-qe/console/-/blob/main/frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/cloudinit/vnc-login.spec.ts

lkladnit commented:
/retest

lkladnit commented:
/unhold

lkladnit commented:
/retest

lkladnit commented:
/retest

lkladnit commented:
/retest

lkladnit commented:
/retest

lkladnit commented:
/retest

lkladnit commented:
/retest

lkladnit commented:
/retest

lkladnit commented:
/retest

lkladnit commented:
/retest

gouyang commented on frontend/packages/kubevirt-plugin/integration-tests-cypress/views/wizard.ts:
```diff
@@ -57,7 +57,13 @@ const fillBootSource = (
     default: {
-      break;
+      // break;
+      cy.get(wizardView.sourceURL).type(provisionSource.getSource());
```
The test failed because here. Suggest to change the code a bit:
```  switch (provisionSource.getDescription()) {
    case 'URL': {
      cy.get(wizardView.sourceURL).type(provisionSource.getSource());
      if (pvcSize) {
        cy.get(wizardView.pvcSize)
          .clear()
          .type(pvcSize);
      }
      break;
    }
    case 'Clone': {
      cy.get('body').then(($body) => {
        if ($body.find(wizardView.clonePVCNSDropdown).length) {
          cy.get(wizardView.clonePVCNSDropdown).click();
          cy.get(wizardView.projectNS(pvcNS)).click();
          cy.contains(wizardView.clonePVCDropDown, 'Select Persistent Volume Claim').click({
            force: true,
          });
          cy.get(wizardView.pvcName(pvcName)).click();
        } else {
          cy.get(wizardView.pvcNSDropdown).click();
          cy.get(wizardView.projectNS(pvcNS)).click();
          cy.get(wizardView.pvcDropdown).click();
          cy.get(wizardView.pvcName(pvcName)).click();
        }
      });
      break;
    }
    case 'Registry': {
      cy.get(wizardView.sourceRegistry)
        .clear()
```

lkladnit replied:
done

lkladnit commented:
/retest

lkladnit commented:
@yaacov @gouyang, can someone look at it and lgtm?

yaacov commented:
@lkladnit you gtot `frontend/packages/console-dynamic-plugin-sdk/package-lock.json` into this PR i don't see that it's connected to any other changes, I assume it's here by mistake, can you remove it ?

lkladnit commented:
removed

> @lkladnit you gtot `frontend/packages/console-dynamic-plugin-sdk/package-lock.json` into this PR i don't see that it's connected to any other changes, I assume it's here by mistake, can you remove it ?

removed

lkladnit commented:
@yaacov should be ok now

yaacov commented:
/lgtm

gouyang commented:
@lkladnit there are 3 tests failed to run in downstream, please check:
https://main-jenkins-csb-cnvqe.apps.ocp-c1.prod.psi.redhat.com/view/Test%20Jobs/view/4.9/job/test-kubevirt-console-t1-cnv-4.9-GOUYANG/94/console

lkladnit commented:
/retest

gouyang commented:
/retest

yaacov commented:
@lkladnit it needs a tracking BZ, o/w it will not get merged

lkladnit commented:
/retest

lkladnit commented:
/lgtm

gouyang commented:
/lgtm

gouyang commented:
@lkladnit The tests in tab/console.spec.ts is failed in this PR, I think you should not touch the code in views/console.ts as it's not used in the tests in this PR.

lkladnit commented:
The new tests mover to [tier 2](https://gitlab.cee.redhat.com/cnv-qe/console/-/merge_requests/1). 
Closing this as redundant
